src/pip/_internal/req/constructors.py

```diff
@@ -74,6 +74,14 @@ def _strip_extras(path):
     return path_no_extras, extras
 
 
+def convert_extras(extras):
+    # type: (Optional[str]) -> Set[str]
+    if extras:
+        return Requirement("placeholder" + extras.lower()).extras
+    else:
+        return set()
```

Comment:
This can follow the pattern used in `with_source()`:

```python
if not extras:
    return set()
return Requirement(...)
```

Reply:
Thanks, updated in #7043.